src/parser.ts

```diff
@@ -393,14 +393,24 @@ export class Parser {
     return modifiers;
   }
 
-  public getParameterInfo(callSignature: ts.Signature) {
+  public getParameterInfo(callSignature: ts.Signature): Method['params'] {
     return callSignature.parameters.map(param => {
+      const paramType = this.checker.getTypeOfSymbolAtLocation(
+        param,
+        param.valueDeclaration
+      );
+      const optionalParam = !!this.checker.symbolToParameterDeclaration(param)!.questionToken;
```

Comment:
This seems like it could crash if `symbolToParameterDeclaration` returned null. Would it be better to write this in a way that avoids the non-null assertion operator?
```
const parameterDeclaration = this.checker.symbolToParameterDeclaration(param);
const optionalParam = !!(parameterDeclaration && parameterDeclaration.questionToken);
```

Reply:
Completely agree, especially as we can see lot of these kind of bugs lately.

Reply:
Is it possible to have a param symbol with a param declaration? i.e. didn't we just read it out?

This crash would be possible if we were synthesizing params using the compiler API, but since we're necessarily reading them from declarations...

Happy to add it but I didn't use the assertion thoughtlessly ;)

Reply:
That's true but I think it's safer anyway to avoid the assertion. There could be other reasons that it would return null that we don't know about, or it might return null unexpectedly in a future version typescript.

Reply:
Added